qutebrowser/config/configtypes.py

```diff
@@ -260,6 +260,37 @@ def validate(self, value):
             raise configexc.ValidationError(value, "items may not be empty!")
 
 
+class FlagList(List):
+
+    """Base class for a list setting that contains one or more flags.
+
+    Lists with duplicate flags are invalid and each item is checked against
+    self.valid_values (if not empty).
+    """
+
+    def validate(self, value):
+        self._basic_validation(value)
+        if not value:
+            return
+
+        vals = self.transform(value)
+        if None in vals and not self.none_ok:
+            raise configexc.ValidationError(
+                value, "May not contain empty values!")
+
+        # Check for duplicate values
+        if len(set(vals)) != len(vals):
+            raise configexc.ValidationError(
+                value, "List contains duplicate values!")
+
+        # Check if each value is valid, ignores None values
+        set_vals = set(val for val in vals if val)
+        if (self.valid_values and self.valid_values.values and
```

Comment:
I think this should just be `self.valid_values is not None` - check against `None` because that's the usual thing to do if a value is sometimes `None`, and I don't think the other check is needed (why would `valid_values.values` ever be falsey?)